### PR TITLE
Move anonymous download to authentication configuration section

### PR DESCRIPTION
**Pull request by Jonkno**
- The anonymous download section was moved to the authentication configuration section as a subitem
- Messages/item description was updated

### CODE REVIEW

**Comment by kaizimmerm** [comment]
A description would be helpful. Thx.

File: hawkbit-ui/src/main/resources/messages_de.properties [hunk 1]
```diff
@@ -159,6 +159,7 @@ label.tag.name = Tag name
 label.configuration.auth.header = Allow targets to authenticate via a certificate authenticated by an reverse proxy
 label.configuration.auth.gatewaytoken = Allow a gateway to authenticate and manage multiple targets through a gateway security token
 label.configuration.auth.targettoken = Allow targets to authenticate directly with their target security token
+label.configuration.anonymous.download = Allow targets to download anonymous
```
Comment (michahirsch) [inline hawkbit-ui/src/main/resources/messages_de.properties]: The label sentence must be `Allow targets to download anonymously` but I guess I would change the sentence more like `Allow targets to download artifacts without security credentials` @kaizimmerm  would you think?

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+@ViewScope
+public class AnonymousDownloadAuthenticationConfigurationItem extends AbstractAuthenticationTenantConfigurationItem {
+
+    private static final long serialVersionUID = 1L;
+
+    private boolean configurationEnabled = false;
+    private boolean configurationEnabledChange = false;
+
+    @Autowired
+    private I18N i18n;
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java]: move I18n + getter in superclass and remove the attirbute in all subclasses

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Autowired
+    public AnonymousDownloadAuthenticationConfigurationItem(
+            final TenantConfigurationManagement tenantConfigurationManagement) {
+        super(TenantConfigurationKey.ANONYMOUS_DOWNLOAD_MODE_ENABLED, tenantConfigurationManagement);
+    }
+
+    @PostConstruct
+    public void init() {
+
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java]: remove empty lines

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        super.init(i18n.get("label.configuration.anonymous.download"));
+        configurationEnabled = isConfigEnabled();
+
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java]: remove empty lines

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @Override
+    public void configEnable() {
+
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java]: remove empty line

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        configurationEnabled = false;
+    }
+
+    @Override
+    public void save() {
+        if (!configurationEnabledChange) {
+            return;
+        }
+        getTenantConfigurationManagement().addOrUpdateConfiguration(getConfigurationKey(), configurationEnabled);
+
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java]: remove empty line

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @Override
+    public void undo() {
+
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java]: remove empty line

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        configurationEnabledChange = false;
+        configurationEnabled = getTenantConfigurationManagement()
+                .getConfigurationValue(getConfigurationKey(), Boolean.class).getValue();
+
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java]: remove empty line

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java]: remove empty line

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        super.init(i18n.get("label.configuration.anonymous.download"));
+        configurationEnabled = isConfigEnabled();
+
+    }
+
+    @Override
+    public void configEnable() {
+
+        if (!configurationEnabled) {
+            configurationEnabledChange = true;
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java]: you can set the variable direct configurationEnabledChange  = !configurationEnabled

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+        }
+        configurationEnabled = true;
+    }
+
+    @Override
+    public void configDisable() {
+        if (configurationEnabled) {
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/tenantconfiguration/authentication/AnonymousDownloadAuthenticationConfigurationItem.java]: you can set the variable directly

**Pull request merged**